Src/AutoNSubstitute/NSubstituteVirtualMethodsCommand.cs

```diff
@@ -197,6 +197,15 @@ public void ReturnsUsingContext<T>(MethodInfo methodInfo)
                 });
             }
 
+            private T Resolve<T>()
+            {
+                var value = Context.Resolve(typeof (T));
+                if (value is OmitSpecimen)
+                    return default(T);
```

Comment:
Are we sure that this is the desired behaviour?

For all reference types, `default(T)` means `null`. However, `OmitSpecimen` isn't the same as `null`. If the `Context` had found that `null` was the appropriate value to return, it would have returned `null`.

`OmitSpecimen` means that the caller should forget about the value - most of the cases by *not* assigning a value to a property or field. That's not the same as assigning `null` to a property or field.

Since this proposed change directly influences `ReturnsUsingContext`, which is a public function, this concerns me.

I also wonder why requesting a `Type` would return `OmitSpecimen`, which is a return value I'd mostly expect to see if a `PropertyInfo` or `FieldInfo` was requested...